core-bundle/src/Filesystem/FilesystemItem.php

```diff
@@ -156,10 +156,21 @@ public function getFileSize(): int
         return $this->fileSize ?? 0;
     }
 
-    public function getMimeType(): string
+    public function getMimeType(bool $throwOnError = true): string
```

Comment:
Isn‘t this considered an anti-pattern? You should always throw the exception and wrap the call in a try-catch block instead.

Reply:
Well, by default this is the case: if you use items in your code and you rely on having a mime type, you can use `getMimeType()` and the exception will bubble up. But if the value is optional, `getMimeType(false)` is much more convenient. 

The latter is also crucial when outputting items in templates, e.g. to add a custom icon class depending on the type:
```twig
<span class="icon icon--{{ item.mimeType(false) }}">{{ item.name }}</span>
```

Reply:
How about a `$default` parameter, and the method only throws if no default is provided? I think this concept is pretty common?

```suggestion
    public function getMimeType(string|null $default = null): string
```

Reply:
Good idea, implemented in 780b0f34d3bbf7403b525e80201997da53c2f2d7.